File: src/main/java/io/playground/orderservice/application/order/port/OrderPersistencePort.java
package io.playground.orderservice.application.order.port;

import io.playground.orderservice.domain.order.Order;

import java.util.List;
import java.util.Optional;

public interface OrderPersistencePort {
    Optional<Order> findByExternalId(String externalId);

    Order save(Order order);

    boolean updateStatusByExternalId(String externalId,
                                     Order.OrderStatus status,
                                     List<Order.OrderStatus> beforeStatuses);
}

File: src/main/kotlin/io/playground/orderservice/application/order/port/OrderPersistencePort.kt
package io.playground.orderservice.application.order.port

import io.playground.orderservice.domain.order.Order
import java.util.Optional

@JvmSuppressWildcards
interface OrderPersistencePort {
    fun findByExternalId(externalId: String): Optional<Order>

    fun save(order: Order): Order

    fun updateStatusByExternalId(
        externalId: String,
        status: Order.OrderStatus,
        beforeStatuses: List<Order.OrderStatus>,
    ): Boolean
}
